src/main/resources/schema/provider.definition.schema.v1.json

```diff
@@ -9,23 +9,46 @@
             "pattern": "^https://[0-9a-zA-Z]([-.\\w]*[0-9a-zA-Z])(:[0-9]*)*([?/#].*)?$",
             "maxLength": 4096
         },
-        "handlerDefinition": {
+        "cudHandlerDefinition": {
```

Comment:
i feel like asyncHandlerDefinition vs syncHandlerDefinition might be a better name. Also should probably update the description on the object in either case so that they are distinguishable

Reply:
I'll update the description, not sure if async vs sync might be confusing though because a CUD handler could be sync or async, depending on how quickly it finishes

Reply:
Need to leave this as it was. `handlerDefinition` is meant to be an abstract action concept. `Read` and `List` operations don't _need_ to define a timeout, but they could and it would still be valid. The `timeoutInMinutes` property should be optional anyway.

Reply:
my idea here was that `Read` and `List` should not be able to define a timeout, and that CUD operations were meant to be async (though as we know could possibly finish synchronously), where as read and list were meant to be synchronous only.  In which case, a timeout would not make sense.  I think its a very bad experience to have an async describe call.  Keep in mind also that this is the two way door decision and could always be changed back later, whereas if we start letting customers define Read and List timeouts (which aren't used in any way) then we have to support that forever

Reply:
Async or Sync response is an orthogonal concern to timeouts. Both can have a maximum wait time. Nothing in the schema indicates the calling mode for CRUDL operations. What I don't want is to start changing the schema to specialise the way handler actions are defined. These are _designed_ in the schema to be descriptions of arbitrary actions that can be taken on a type. They have a name, but they are just actions. It's up to the caller to understand the interface and the contract, including whether to call and wait synchronously, what the action payloads are, etc. The schema shouldn't attempt to start trying to define that interface and specialise each operation's definition - that really is a one-way door. 

On top of that, changing this now invalidates every existing schema that has already been published. This is not the right change. There is absolutely nothing wrong with having optional timeouts on all handlers which may or may not be honored by the caller for certain operations. You are defining a _maximum_ timeout, we can wait less as we choose.